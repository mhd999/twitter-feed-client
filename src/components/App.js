// @flow

import React, { Component } from 'react';
import TweetsList from './TweetsList';
import { API_URL } from '../config';

class App extends Component {
    constructor() {
        super();
        this.state = {
            tweets: [],
        };
    }
    state : {
        tweets: any[]
    };

    componentDidMount() {
        fetch(`${API_URL}/dashboard`)
          .then(response => {
            response.json().then(data => {
            if (response.status !== 200) {
                throw new Error('success, no data');
            } else {
                this.setState({ tweets: data.tweets });
             }
            });
        });
    }

    render() {
        return(
            <div id='app'>
                {this.state.tweets.length > 0 && 
                    <TweetsList tweets={this.state.tweets} /> 
                }
            </div>
        );
    }
}

export default App;