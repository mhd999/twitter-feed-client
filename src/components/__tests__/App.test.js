import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from '../App';

describe('<App />', () => {
  it('renders children', () => {
    const component = shallow(
      <App  />
    );
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
