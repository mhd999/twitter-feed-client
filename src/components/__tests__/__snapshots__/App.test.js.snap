exports[`<App /> renders children 1`] = `
<div
  id="app" />
`;
